optaplanner-core/src/main/java/org/optaplanner/core/impl/solver/DefaultSolverJob.java

```diff
@@ -73,15 +85,16 @@ public DefaultSolverJob(
         }
         this.solver = (DefaultSolver<Solution_>) solver;
         this.problemFinder = problemFinder;
+        this.bestSolutionConsumer = bestSolutionConsumer;
         this.finalBestSolutionConsumer = finalBestSolutionConsumer;
         this.exceptionHandler = exceptionHandler;
         solverStatus = SolverStatus.SOLVING_SCHEDULED;
         terminatedLatch = new CountDownLatch(1);
         solverStatusModifyingLock = new ReentrantLock();
     }
 
-    public void setFuture(Future<Solution_> future) {
-        this.future = future;
+    public void setFinalBestSolutionFuture(Future<Solution_> finalBestSolutionFuture) {
+        this.finalBestSolutionFuture = finalBestSolutionFuture;
     }
```

Comment:
This comes from the `SolverManager` and represents the `SolverJob` itself running in the `Executor`. It may be canceled or waited for - see `getFinalBestSolution`.